pkg/validate/validate.go

```diff
@@ -68,8 +65,21 @@ func validateContainerService(c *api.OpenShiftManagedCluster, externalOnly bool)
 }
 
 func validateUpdateContainerService(cs, oldCs *api.OpenShiftManagedCluster, externalOnly bool) (errs []error) {
-	if !reflect.DeepEqual(cs, oldCs) {
```

Comment:
@jim-minter not sure why these changes were reverted in https://github.com/openshift/openshift-azure/pull/238 - https://github.com/openshift/openshift-azure/pull/181 was a more reasonable approach to validate updates vs invalidating everything.

Reply:
There are a couple of extra fields that we need to gate apart from the current, `vmSize`, `publicHostname`, and `osType`. I can invalidate changes to those too, error messages are going to be friendlier.

Reply:
Opted for your suggestion as it was much less code, better for review and the error message is ok.

Reply:
It's much uglier though because we need to figure out a difference in counts in two different places: 1. validation (here) and 2. before apply the ARM template. So mutating counts in validation is a no-no. For now, I am copying the counts back after running deep equal. Eventually we should have a real deep copy for OpenShiftManagedCluster.